core/primitives/src/version.rs

```diff
@@ -176,11 +176,15 @@ pub const PEER_MIN_ALLOWED_PROTOCOL_VERSION: ProtocolVersion = STABLE_PROTOCOL_V
 const STABLE_PROTOCOL_VERSION: ProtocolVersion = 55;
 
 /// Version used by this binary.
-#[cfg(not(feature = "nightly_protocol"))]
+#[cfg(all(not(feature = "nightly_protocol"), not(feature="shardnet")))]
```

Comment:
I would reach for `cfg_if!` at this point. It is a very small crate and I’m ~100% confident it is already in our dependency graph _somewhere_.